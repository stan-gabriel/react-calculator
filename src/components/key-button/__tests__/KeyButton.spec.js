import React from 'react';
import { shallow } from 'enzyme';
import Button from '../KeyButton';

describe('KeyButton component', () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<Button />)));

	it('should match the snapshot', () => expect(wrapper).toMatchSnapshot());

	it('should render a <button />', () => {
		expect(wrapper.find('button').length).toEqual(1);
	});
});
