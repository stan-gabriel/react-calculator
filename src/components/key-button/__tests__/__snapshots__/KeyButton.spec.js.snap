// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyButton component should match the snapshot 1`] = `
<button
  className="class KeyButton extends
/* istanbul ignore next */
(
/* istanbul ignore next */
_react.
/* istanbul ignore next */
default.Component) {
  render() {
    /* istanbul ignore next */
    cov_175xnr8u4p.f[0]++;

    const _ref = (cov_175xnr8u4p.s[0]++, this.props),
          value = _ref.value,
          colour = _ref.colour,
          onClick = _ref.onClick;

    const className =
    /* istanbul ignore next */
    (cov_175xnr8u4p.s[1]++, \`\${KeyButton} \${colour}\`);

    /* istanbul ignore next */
    cov_175xnr8u4p.s[2]++;
    return (
      /* istanbul ignore next */
      _react.default.createElement(
      /* istanbul ignore next */
      \\"button\\",
      /* istanbul ignore next */
      {
        className: className,
        onClick: onClick,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 9
        },
        __self: this
      }, value)
    );
  }

} undefined"
/>
`;
